modules/apm/src/main/plugin-metadata/entitlement-policy.yaml

```diff
@@ -1,4 +1,8 @@
 org.elasticsearch.telemetry.apm:
   - create_class_loader
+  - write_all_properties
```

Comment:
Alternatively this could be a prefix wildcard entitlement as discussed in Slack to only allow `elastic.apm.*`.
However, on the other hand we're planning to remove the agent, so this would be hopefully obsolete rather soon.

Reply:
The thing I do not really like about this is that this is the only place we need WriteAll in plugins. I'd rather avoid having WriteAll exposed.
I like what you proposed on slack, list the keys you can actually set, since we only allow a subset. It's true that we need to be very good at keeping them in sync, but since this is provisional, maybe we can do this for now? And add a big comment/disclaimer on `PERMITTED_AGENT_KEYS`? I'd prefer that.

Reply:
Though, happy to be convinced that we can handle the risk and go with a fixed list...

Reply:
> the server crashes really badly if this is out of sync

Do we/can we have tests which could cover this case? So we can discover immediately if this happens? (e.g. a IT test in the APM plugin trying to set all properties in `PERMITTED_AGENT_KEYS`?)

Reply:
The line below adds a bit of uncertainty, but if the workaround / exclusion is removed the IT test should give that confidence. I'll give this a go 👍 
https://github.com/elastic/elasticsearch/blob/c760d73c55d7c8dac6837aa75b13ebfd39a22471/modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java#L238-L242

Reply:
That's great! If that turns out well (an IT test can cover our back), I'd say we go with the list of explicit property names.
If not... well, we can keep it as is for the moment, I suppose.

Reply:
ok, fairly confident this will catch any issue 👍